src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -238,13 +238,41 @@ public void Translate(object sender, FailedRequestEventArgs args)
             }
         }
 
-        public event Action<Topology> TopologyUpdated;
-
         private void OnFailedRequestInvoke(string url, Exception e)
         {
             _onFailedRequest?.Invoke(this, new FailedRequestEventArgs(_databaseName, url, e));
         }
 
+        private event EventHandler<TopologyUpdatedEventArgs> _onTopologyUpdated;
+
+        public event EventHandler<TopologyUpdatedEventArgs> OnTopologyUpdated
+        {
+            add
+            {
+                lock (_locker)
+                {
+                    _onTopologyUpdated += value;
```

Comment:
This should be thread safe, no need for locking

Reply:
Done: https://github.com/ravendb/ravendb/commit/b25e4814b1e65b27e456fb44c2f26acaf94bd263